### PR TITLE
Use BGRT Plymouth theme

### PR DESCRIPTION
**Pull request by meisenzahl**
Closes https://github.com/elementary/plymouth-theme/issues/3

### CODE REVIEW

**Comment by meisenzahl** [comment]
When updating the initramfs with `sudo update-initramfs -u -k all` I get the following warning:

```
W: plymouth module (/usr/lib/x86_64-linux-gnu/plymouth//two-step.so) missing, skipping that theme.
```

So the theme is not installed.

## Ubuntu 20.041

This works with Ubuntu 20.04.1. Plymouth related installed packages:

```
ii  libplymouth5:amd64                         0.9.4git20200323-0ubuntu6           amd64        graphical boot animation and logger - shared libraries
ii  plymouth                                   0.9.4git20200323-0ubuntu6           amd64        boot animation, logger and I/O multiplexer
ii  plymouth-label                             0.9.4git20200323-0ubuntu6           amd64        boot animation, logger and I/O multiplexer - label control
ii  plymouth-theme-spinner                     0.9.4git20200323-0ubuntu6           amd64        boot animation, logger and I/O multiplexer - spinner theme
ii  plymouth-theme-ubuntu-text                 0.9.4git20200323-0ubuntu6           amd64        boot animation, logger and I/O multiplexer - ubuntu text theme
```

## elementary OS 6 daily

```
ii  libplymouth5:amd64                         0.9.4git20200323-0ubuntu6.2            amd64        graphical boot animation and logger - shared libraries
ii  plymouth                                   0.9.4git20200323-0ubuntu6.2            amd64        boot animation, logger and I/O multiplexer
ii  plymouth-label                             0.9.4git20200323-0ubuntu6.2            amd64        boot animation, logger and I/O multiplexer - label control
ii  plymouth-theme-elementary                  5.1.2-0~348~daily~ubuntu6.0.1          all          changes the plymouth theme to elementary OS
ii  plymouth-theme-elementary-text             5.1.2-0~348~daily~ubuntu6.0.1          all          Dummy transitional package
```

**Comment by meisenzahl** [comment]
## Update

Missing library is provided by `plymouth-theme-spinner`.

**Comment by meisenzahl** [comment]
I copied all files from Ubuntu's theme and only slightly modified them. (`watermark.png` and `elementary.plymouth`)

Bildschirmfoto von 2020-11-15 18-11-52

**Comment by meisenzahl** [comment]
@cassidyjames the animated spinner is a series of PNGs. How is elementary's spinner created? Pure CSS?

Could we export it as single PNGs?

We have to play a bit with size of elementary's logo. Ubuntu's logo is 248 x 49. I resized elementary's logo to 64 x 64.

**Comment by cassidyjames** [comment]
@meisenzahl honestly I'm wondering if we just drop the elementary logo from here. This screen is typically short-lived, and I don't want to over-brand it with competing logos.

Regarding the spinner, there is [@elementary/icons/status/symbolic/process-working-symbolic](https://github.com/elementary/icons/blob/master/status/symbolic/process-working-symbolic.svg), which we could recolor to white and export different states. How many images do you need?

@danrabbit thoughts on both of the above?

**Comment by meisenzahl** [comment]
@cassidyjames I would also have no problem removing the elementary logo. The screen looks a bit overloaded.

We need one image for each frame of the animation. We might have to resize it a bit. The Ubuntu images are 38 x 38.

**Comment by meisenzahl** [comment]
Peek 2020-11-16 12-23

**Comment by meisenzahl** [comment]
Peek 2020-11-16 16-39

**Comment by cassidyjames** [comment]
@meisenzahl I really like how minimal this is. We should probably decide/test some things before merging:

- [ ] What happens when the hardware doesn't have an image? Is that a possibility?
- [ ] _Are_ we okay leaving the OS branding out here? I like how clean it looks, personally.
- [ ] How does this look/work for things like firmware updates (the original reason we were looking at a bgrt theme)? How do we test that?
- [ ] How does it look for encryption?

Anything else I'm missing?

**Comment by meisenzahl** [comment]
> ## Update
> 
> Missing library is provided by `plymouth-theme-spinner`.
> 
> Reference: https://packages.ubuntu.com/focal-updates/amd64/plymouth-theme-spinner/filelist

How do we package it? We need `two-step.so` which is provided by `plymouth-theme-spinner`. Which is the Ubuntu Plymouth theme.

**Comment by cassidyjames** [comment]
@meisenzahl I really like how minimal this is. We should probably decide/test some things before merging:

- [x] What happens when the hardware doesn't have an image? Is that a possibility?
- [ ] _Are_ we okay leaving the OS branding out here? I like how clean it looks, personally.
- [ ] How does this look/work for things like firmware updates (the original reason we were looking at a bgrt theme)? How do we test that?
- [ ] How does it look for encryption?
- [ ] Are there assets we're missing, like text entries or progress bars or anything?

Anything else I'm missing?

**Comment by meisenzahl** [comment]
I created an Merge Request for Plymouth:

[Use fallback image if BGRT is not supported
](https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/125)

**Comment by meisenzahl** [comment]
### Firmware upgrade

Bildschirmfoto von 2020-11-17 17-11-33

### System upgrade

Bildschirmfoto von 2020-11-17 17-11-46

### Updates

Bildschirmfoto von 2020-11-17 17-12-03

### Password prompt

Bildschirmfoto von 2020-11-17 17-14-08

**Comment by meisenzahl** [comment]
Regarding the password prompt we could provide our own

- `entry.png`
- `lock.png`

**Comment by cassidyjames** [comment]
@meisenzahl I updated my comment above, but I think the main thing blocking this now is documenting how to install/test it on elementary OS 6.

File: elementary/elementary.plymouth [hunk 1]
```diff
@@ -1,8 +1,56 @@
...
 Name=elementary
-Description=A theme that features a blank background with a logo.
-ModuleName=script
+Description=elementary theme using the ACPI BGRT graphics as background
+ModuleName=two-step
 
-[script]
+[two-step]
+Font=Inter 12
+TitleFont=Inter Light 30
```
Comment (cassidyjames) [inline elementary/elementary.plymouth:8]: This more closely matches our Granite Welcome styles with h1/h2.

```suggestion
Font=Inter Light 24
TitleFont=Inter Bold 32
```

**Comment by davidmhewitt** [comment]
I can confirm that the debian packaging works as expected.

When installing the deb built from this branch, it correctly pulls in the `plymouth-theme-spinner` package from the Ubuntu repos, but the elementary theme takes precedence and is set as the active plymouth theme.

Even when forcibly re-installing the `plymouth-theme-spinner` package, the elementary theme stays the default.

**Comment by cassidyjames** [comment]
Updating with some screenshots from my latest changes and testing. Also, note that at least on my HiDPI Dell Precision 5530, testing in the session will show a double-sized logo, but during an actual boot it is correctly sized.

**Comment by davidmhewitt** [comment]
While I'm happy with these changes technically and have tested that the debian packaging works, the current `plymouth-theme-elementary` package is generated from the `elementary-default-settings` source package.

We can see the source of the plymouth theme here:
https://github.com/elementary/default-settings/tree/master/plymouth

I think if we set up a launchpad recipe for this repo, and updated the debian changelog here so that the version of this was newer than the one from default settings (currently 5.1.2), this would supersede the one from default-settings. Then we could remove that from the packaging there.

I feel it makes sense for this to be in its own separate repo, but it's just a bit fiddly to swap the debian package over.

**Comment by cassidyjames** [comment]
@danrabbit and/or @tintou could you look into that? ^

**Comment by tintou** [comment]
I've updated the debian packaging and added a breaks/replaces of the current version of elementary-default-settings.
This package can only be installed with an upgraded elementary-default-settings.

**Review by tintou — APPROVED** [review]

**Comment by meisenzahl** [comment]
I suggest that we first merge https://github.com/elementary/plymouth-theme/pull/7.

This way we do not pollute the Git history with the png files.

@cassidyjames do we want to transfer the explanations in the README to the Wiki?

**Comment by cassidyjames** [comment]
@meisenzahl I'm indifferent; the README is fine to me

**Comment by meisenzahl** [comment]
> I created a Merge Request for Plymouth:
> 
> [Use fallback image if BGRT is not supported ](https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/125)

The feature was recently merged.

I created an issue [to deploy the latest version of Plymouth](https://github.com/elementary/os-patches/issues/150).

@cassidyjames while we are here: can you provide a version of elementary's logo as fallback image for ARM based devices?

It has to be located in `elementary/bgrt-fallback.png`

**Comment by cassidyjames** [comment]
@meisenzahl reusing https://github.com/elementary/plymouth-theme/blob/master/elementary/logo.png seems to make the most sense to me

**Comment by cassidyjames** [comment]
@meisenzahl I think I'd rather get this merged sooner instead of it being blocked forever.

**Pull request merged**